hass_client/client.py

```diff
@@ -212,6 +232,10 @@ async def send_command(self, command: str, **kwargs: dict[str, Any]) -> CommandR
         await self._send_json_message(message)
         try:
             return await future
+        except CancelledError as e:
+            if len(e.args) > 0:
+                # Raise the inner exception
+                raise e.args[0] from e
```

Comment:
what is this ?

It is not very common to catch the CancelledError, why are you doing this like this ?

Reply:
The websocket connection shuts down when a TooLarge error message is returned, which means we experience the failure as a cancellation. But we need to know what actually caused it to shutdown, which is packaged as the first parameter of `e.args` - since from SendCommand we'd like to get the actual exception, not just "everything was aborted".

The relevant usage is here:

```python
    async def send_retryable_command(self, command: str, **kwargs: dict[str, Any]) -> CommandResultData:
        """Send a command to the HA websocket and return response. Retry on failure."""
        while True:
            try:
                return await self.send_command(command, **kwargs)
            except ConnectionFailedDueToLargeMessage:
                LOGGER.debug("Connection failed due to large message - waiting for reconnect and then retrying")
                await self.wait_for_connection()
```

where we want to either keep raising the exception (because it's not something known harmless), or handle it because we actually know what it is (ConnectionFailedDueToLargeMessage). Otherwise calling code doesn't know why the connection was ended.

Of note: this catch *could* be other things, but I don't know what they should be at the moment (a reasonable one would be something like "ConnectionLostRetriesExceeded" or something.